Tests/ComposableArchitectureTests/DebugTests.swift

```diff
@@ -216,5 +216,20 @@
       let store = TestStore(initialState: 0, reducer: DebuggedReducer()._printChanges())
       await store.send(true) { $0 = 1 }
     }
+    
+    @MainActor
+    func testDebugReducerInPreview() async {
+      struct DebuggedReducer: ReducerProtocol {
+        typealias State = Int
+        typealias Action = Bool
+        func reduce(into state: inout Int, action: Bool) -> EffectTask<Bool> {
+          state += action ? 1 : -1
+          return .none
+        }
+      }
+      let store = TestStore(initialState: 0, reducer: DebuggedReducer()._printChanges())
+      store.dependencies.context = .preview
```

Comment:
@stephencelis We can't really test the behavior directly, as there is no way to override at distance the default `TextOuputStream` from `print()` to my knowledge. Maybe there are some way to read stdout, but that seems overkill.

Switching this value to `.live` shows that something is printed, and setting it back to `.preview` shows that nothing is printed, like for the test above that runs in `.test` context.

Reply:
Ah yeah. We should prob allow more customization in the future, but the API is underscored and barebones at the moment.

Reply:
> there is no way to override at distance the default TextOuputStream from print()

@tgrapperon Back when you experimented with #1234 I had a go at just this :P

You can redirect stdout to `Pipe` using [dup2](https://man7.org/linux/man-pages/man2/dup.2.html)

It took a little bit of trial and error but I ended up with something that worked okay. I haven't used it though. It was just a proof of concept.

[iampatbrown/ConsoleView.swift](https://gist.github.com/iampatbrown/3034ec9756a6cdc3b9dd25d9784f3486)

Reply:
@iampatbrown Super interesting! I'll give it a spin if I rework on #1234 idea. I'm currently using my `Logger` abstraction to duplicate prints to the UI console, but your approach could make things even simpler.